sklearn/cluster/_kmeans.py

```diff
@@ -2144,17 +2144,27 @@ def fit(self, X, y=None, sample_weight=None):
         self._n_since_last_reassign = 0
 
         n_steps = (self.max_iter * n_samples) // self._batch_size
-
+        sample_weight = sample_weight / np.sum(sample_weight)
+        sample_weight_minibatch_step = np.ones(self._batch_size).astype(X.dtype)
```

Comment:
We need to keep an unnormalized version of the sample weights because they're used afterwards to compute the final inertia which scales with the sample weights.

```suggestion
        normalized_sample_weight = sample_weight / np.sum(sample_weight)
        unit_sample_weight = np.ones_like(sample_weights, shape=(self._batch_size,))
```